packages/react-sdk/src/useRiverConnection.ts

```diff
@@ -4,18 +4,26 @@ import type { ethers } from 'ethers'
 import { connectRiverWithBearerToken, signAndConnect } from './connectRiver'
 import { useRiverSync } from './internals/useRiverSync'
 
+type RiverConnectConfig = Omit<SyncAgentConfig, 'context' | 'onTokenExpired'>
 export const useRiverConnection = () => {
     const [isConnecting, setConnecting] = useState(false)
     const river = useRiverSync()
 
     const connect = useCallback(
-        async (signer: ethers.Signer, config: Omit<SyncAgentConfig, 'context'>) => {
+        async (signer: ethers.Signer, config: RiverConnectConfig) => {
             if (river?.syncAgent) {
                 return
             }
-
+            const mergedConfig = {
+                ...config,
+                ...river?.config,
+                onTokenExpired: () => {
+                    river?.config?.onTokenExpired?.()
+                    river?.setSyncAgent(undefined)
```

Comment:
Unsure if we want this to the default case - lets see

Reply:
my token expired in marbella, and the app was still usable for reads, you just can't post or create streams. 
i think this will be jarring because it will seem like your app is working, then when you go to post a message everything will just reset to nothing yeah? A modal would probably be better, but we can improve it as we go.

Reply:
>  A modal would probably be better

the react-sdk will expose the `onTokenExpired` callback, so you can pop a modal or other UI element to indicate this.

But now we have two situations:
1. let the dev control when the sync agent should reset
2. we control when the sync agent should reset

I need to think more about this